core-implementation/src/main/java/gg/solarmc/loader/impl/IcarusLauncher.java

```diff
@@ -98,7 +99,8 @@ public Icarus launch(SolarDataConfig.DatabaseCredentials credentials) {
 				.load();
 		flyway.migrate();
 
-		ExecutorService executor = executorServiceFactory.newFixedThreadPool(dataSource.getMaximumPoolSize());
+		ExecutorService executor = executorServiceFactory.newFixedThreadPool(
+				dataSource.getMaximumPoolSize(), new DataLoaderThreadFactory());
```

Comment:
what is the point of passing a thread factory in there? (just asking cuz i can't see the code before or after this)

Reply:
Also isn't there somewhere else we can get threads from besides manually initializing them? Maybe hikari since it already has a pool of them?

Reply:
If you don't specify a thread factory, the default one is used, which generates names like "pool-1-thread-2". DataLoaderThreadFactory simply changes the name format to "DataLoader-1"

HikariCP has a couple connection pools it uses for its own purposes, but for good reason, it does not expose them to API consumers. They're internal to Hikari - the housekeeper and the connection assassination thread pool.